crypto/ec/ec_lib.c

```diff
@@ -318,6 +318,14 @@ int EC_GROUP_set_generator(EC_GROUP *group, const EC_POINT *generator,
     } else
         BN_zero(&group->cofactor);
 
+    /*-
+     * In FIPS mode ec_precompute_mont_data() always fails: in this case we
+     * skip to the end to clear the group->mont_data pointer and return success,
+     * as there is nothing left to do.
+     */
+    if (FIPS_mode())
```

Comment:
All calls to FIPS_mode elsewhere are guarded with #ifdef OPENSSL_FIPS. It would be appropriate to do same here...

Reply:
Yes, I'm working on it.

I also went further in my analysis, and the patch as it is now has another problem, I'll be pushing a new commit shortly.